### PR TITLE
Okta: Align Okta Plugin Details

### PR DESCRIPTION
**Pull request by smallinsky**
### What

What add details about configured credentials in Okta plugin 

Add HTTPClient into TeleportTestConfig (allowing to mock third party integration calls without relaying on global shared variables)

### CODE REVIEW

File: e
Comment (kopiczko) [inline e:1]: Hmm I can't find 18ac0f

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -6347,6 +6347,20 @@ message PluginOktaSettings {
...
+
+// Defines settings for the SCIM plugin.
+message PluginOktaCredentialsInfo {
+  option (gogoproto.equal) = true;
+  // HasConfiguredSSMSToken is true if the plugin has configured SSMSToken.
+  bool hasConfiguredSSMSToken = 1;
+  // HasConfiguredOauthCredentials is true if the plugin has configured OauthCredentials.
+  bool hasConfiguredOauthCredentials = 2;
+  // HasConfiguredSCIMToken is true if the plugin has configured SCIMToken.
+  bool hasConfiguredSCIMToken = 3;
```
Comment (kopiczko) [inline api/proto/teleport/legacy/types/types.proto:6363]: Shouldn't be snake_case?

File: e
Comment (smallinsky) [reply e:1]: This is leftover from testing. removed

File: api/proto/teleport/legacy/types/types.proto (hunk 1 above)
Comment (smallinsky) [reply api/proto/teleport/legacy/types/types.proto]: Yes, In this file we have mix of CamelCase and snake_case but snake_case where snake_case is suggested for google proto. 
Fixed.

**Review by kopiczko — APPROVED** [review]
The change itself is fine, but it would be good to see how it's used in the context. Can you point to something?

File: api/proto/teleport/legacy/types/types.proto [hunk 2]
```diff
@@ -6347,6 +6347,20 @@ message PluginOktaSettings {
 
   // Sync settings controls the user and access list sync settings for Okta.
   PluginOktaSyncSettings sync_settings = 4;
+
+  // CredentialsInfo contains information about the Okta credentials.
+  PluginOktaCredentialsInfo credentials_info = 5;
+}
+
+// Defines settings for the SCIM plugin.
```
Comment (r0mant) [inline api/proto/teleport/legacy/types/types.proto:6355]: This godoc feels off.

File: api/proto/teleport/legacy/types/types.proto [hunk 3]
```diff
@@ -6347,6 +6347,20 @@ message PluginOktaSettings {
...
+message PluginOktaCredentialsInfo {
+  option (gogoproto.equal) = true;
+  // HasConfiguredSSMSToken is true if the plugin has configured SSMSToken.
+  bool has_configured_ssm_token = 1;
+  // HasConfiguredOauthCredentials is true if the plugin has configured OauthCredentials.
+  bool has_configured_oauth_credentials = 2;
+  // HasConfiguredSCIMToken is true if the plugin has configured SCIMToken.
+  bool has_configured_scim_token = 3;
```
Comment (r0mant) [inline api/proto/teleport/legacy/types/types.proto:6363]: Nit: Can we shorten these?
```suggestion
  // HasSSMSToken is true if the plugin has configured SSMS token.
  bool has_ssm_token = 1;
  // HasOauthCredentials is true if the plugin has configured OAuth 2.0 credentials.
  bool has_oauth_credentials = 2;
  // HasSCIMToken is true if the plugin has configured SCIM token.
  bool has_scim_token = 3;
```

**Review by r0mant — APPROVED** [review]
lgtm with a couple nitpicks

**Pull request merged**